app/src/test/kotlin/compiler/intermediate_form/LivenessTest.kt

```diff
@@ -0,0 +1,275 @@
+        val linearProgram = listOf(
+            Instruction.InPlaceInstruction.Dummy(regsDefined = listOf(reg1)),
+            Instruction.ConditionalJumpInstruction.Dummy("label2"),
+            Label("label1"),
+            Instruction.InPlaceInstruction.Dummy(regsUsed = listOf(reg2)),
+            Instruction.RetInstruction.Dummy(),
+            Label("label2"),
+            Instruction.InPlaceInstruction.Dummy(regsUsed = listOf(reg1)),
+            Instruction.InPlaceInstruction.Dummy(regsDefined = listOf(reg2)),
+            Instruction.UnconditionalJumpInstruction.Dummy("label1")
+        )
+        val expectedLivenessGraphs = Liveness.LivenessGraphs(
+            mapOf(reg1 to setOf(reg2), reg2 to setOf(reg1)),
+            mapOf(reg1 to setOf(), reg2 to setOf())
+        )
+
+        val livenessGraphs = Liveness.computeLiveness(linearProgram)
+        assertEquals(expectedLivenessGraphs, livenessGraphs)
+    }
+
+    @Test
+    fun `interference in loop`() {
+        val linearProgram = listOf(
+            Instruction.InPlaceInstruction.Dummy(regsDefined = listOf(reg1)),
+            Label("loop_start"),
+            Instruction.ConditionalJumpInstruction.Dummy("end_of_loop"),
+            Instruction.InPlaceInstruction.Dummy(regsUsed = listOf(reg1)),
+            Instruction.InPlaceInstruction.Dummy(regsDefined = listOf(reg2)),
+            Instruction.UnconditionalJumpInstruction.Dummy("loop_start"),
+            Label("end_of_loop"),
+            Instruction.RetInstruction.Dummy(),
+        )
+        val expectedLivenessGraphs = Liveness.LivenessGraphs(
+            mapOf(reg1 to setOf(reg2), reg2 to setOf(reg1)),
+            mapOf(reg1 to setOf(), reg2 to setOf())
+        )
+
+        val livenessGraphs = Liveness.computeLiveness(linearProgram)
+        assertEquals(expectedLivenessGraphs, livenessGraphs)
+    }
+
+    @Test
+    fun `copying is not interference`() {
+        val linearProgram = listOf(
+            Instruction.InPlaceInstruction.Dummy(regsDefined = listOf(reg1, reg3)),
+            Instruction.InPlaceInstruction.MoveRR(reg2, reg1),
+            Instruction.InPlaceInstruction.Dummy(regsUsed = listOf(reg1, reg2, reg3)),
+            Instruction.RetInstruction.Dummy()
+        )
+        val expectedLivenessGraphs = Liveness.LivenessGraphs(
+            mapOf(reg1 to setOf(reg3), reg2 to setOf(reg3), reg3 to setOf(reg1, reg2)),
+            mapOf(reg1 to setOf(reg2), reg2 to setOf(reg1), reg3 to setOf())
+        )
+
+        val livenessGraphs = Liveness.computeLiveness(linearProgram)
+        assertEquals(expectedLivenessGraphs, livenessGraphs)
+    }
+
+    @Test
+    fun `copying and interference`() {
+        val linearProgram = listOf(
+            Instruction.InPlaceInstruction.Dummy(regsDefined = listOf(reg1, reg2, reg3)),
+            Instruction.InPlaceInstruction.MoveRR(reg2, reg1),
+            Instruction.InPlaceInstruction.Dummy(regsUsed = listOf(reg1, reg2, reg3)),
```

Comment:
According to my comment above. If I understand correctly, the removal of `reg1` in this line shouldn't change the result and it changes.